transform/macros/repo_event_by_period.sql

```diff
@@ -0,0 +1,12 @@
+{% macro repo_event_by_period(event_type, period) -%}
+   SELECT 
+     repo_id,
+     repo_name,
+     DATE_TRUNC({{period}}, event_date) AS date_{{period}},
+     COUNT(*) AS {{event_type}}_count,
+     SUM({{event_type}}_count) OVER (PARTITION BY repo_id ORDER BY date_{{period}}) AS comul_{{event_type}}_count
+     FROM {{ref("stg_gharchive")}}
+     WHERE event_type = {{event_type}}
```

Comment:
Unless I'm missing sthg, this filter won't work with star and commit. To use these parameters, you'd need to modify your staging table to transform push to commit and watch to star. Alternatively, you could just keep the same nomenclature in your marts pushes and watches.